CDCgov/prime-reportstream
# Fixes authorization bug for `/settings/receivers` list API endpoint

This PR fixes a bug that directed auth to look for `DH` rather than `DHmd_phd`

Test Steps:
1. Ensure your OktaPreview account has only a receiver org present in its groups, no `DHPrimeAdmins`
2. Make a call with that bearer token to the `settings/organizations/{organizationName}/receivers` endpoint
3. Ensure you're authorized to access receivers for that organization.
4. Ensure you're NOT authorized to access other org's receivers.

## Changes
-  Plugs a variable into the auth function

## Checklist

### Testing

### Process

## Linked Issues
- Fixes #issue

## To Be Done
*Create GitHub issues to track the work remaining, if any*
- #issue

## Specific Security-related subjects a reviewer should pay specific attention to
- Does this PR introduce new endpoints?
No
- Does this PR include changes in authentication and/or authorization of existing endpoints?
> **Yes**, it removes an empty string and replaces it with the org name, so the auth process looks for the proper Okta group claim.
- Does this change introduce new dependencies that need vetting?
No
- Does this change require changes to our infrastructure?
No
- Does logging contain sensitive data?
No
- Does this PR include or remove any sensitive information itself?
No

If you answered '_yes_' to any of the questions above, conduct a detailed Review that addresses at least:

- What are the potential security threats and mitigations? Please list the _STRIDE_ threats and how they are mitigated
    - **S**poofing (faking authenticity)
        - Threat _T_, which could be achieved by _A_, is mitigated by _M_
    - **T**ampering (influence or sabotage the integrity of information, data, or system)
    - **R**epudiation (the ability to dispute the origin or originator of an action)
    - **I**nformation disclosure (data made available to entities who should not have it)
    - **D**enial of service (make a resource unavailable)
    - **E**levation of Privilege (reduce restrictions that apply or gain privileges one should not have)
- Have you ensured logging does not contain sensitive data?
- Have you received any additional approvals needed for this change?
